scratchml/models/multilayer_perceptron.py

```diff
@@ -333,6 +350,12 @@ def fit(self, X: np.ndarray, y: np.ndarray) -> None:
                         loss = binary_cross_entropy(batch_y, y_hat, derivative=True)
                     elif self.loss_function == "cross_entropy":
                         loss = cross_entropy(batch_y, y_hat, derivative=True)
+                elif isinstance(self, MLPRegressor):
+                    if self.loss_function == "mse":
+                        loss = 2 * (y_hat - batch_y) / len(batch_y)  # Derivative of MSE w.r.t predictions
+
+                if loss is None:
```

Comment:
Did you try to set loss equal to None? Did the code run? If so, it's better to set that validation in the init function, such as:

```python
# Validate the loss function
        if not self.loss_function:
            if self.loss_function not in self._valid_loss_functions:
                raise ValueError(
                    f"The value for 'loss_function' must be one of {self._valid_loss_functions}."
                )
        else:
            raise ValueError("Loss function computation failed. Check configuration.")
```

If the code did not run and raised a ValueError, you can just delete both lines.

Reply:
Both lines have been removed.